tools/gatk4.wdl

```diff
@@ -0,0 +1,358 @@
+        maxRetries: 1
+    }
+}
+
+task base_recalibrator {
+    meta {
+        description: "Generates recalibration report for base quality score recalibration."
+        external_help: "https://gatk.broadinstitute.org/hc/en-us/articles/360036897372-BaseRecalibratorSpark-BETA"
+        outputs: {
+            recalibration_report: "Recalibration report file"
+        }
+    }
+
+    parameter_meta  {
+        bam: "Input BAM format file on which to recabilbrate base quality scores"
+        bam_index: "BAM index file corresponding to the input BAM"
+        outfile_name: "Name for the output recalibration report."
+        fasta: "Reference genome in FASTA format"
+        fasta_index: "Index for FASTA format genome"
+        dict: "Dictionary file for FASTA format genome"
+        dbSNP_vcf: "dbSNP VCF file"
+        dbSNP_vcf_index: "dbSNP VCF index file"
+        known_indels_sites_VCFs: "List of VCF files containing known indels"
+        known_indels_sites_indices: "List of VCF index files corresponding to the VCF files in `known_indels_sites_VCFs`"
+        memory_gb: "RAM to allocate for task, specified in GB"
+        modify_memory_gb: "Add to or subtract from dynamic memory allocation. Default memory is determined by the size of the inputs. Specified in GB."
+        ncpu: "Number of cores to allocate for task"
+        use_original_quality_scores: "Use original quality scores from the input BAM. Default is to use recalibrated quality scores."
+    }
+
+    input {
+        File bam
+        File bam_index
+        String outfile_name = basename(bam, ".bam") + ".recal.txt"
+        File fasta
+        File fasta_index
+        File dict
+        File dbSNP_vcf
+        File dbSNP_vcf_index
+        Array[File] known_indels_sites_VCFs
+        Array[File] known_indels_sites_indices
+        Int memory_gb = 25
+        Int modify_disk_size_gb = 0
+        Int ncpu = 4
+        Boolean use_original_quality_scores = true
+    }
+
+    Int disk_size_gb = ceil(size(bam, "GB") + 1) * 3 + ceil(size(fasta, "GB")) + modify_disk_size_gb
+    Int java_heap_size = ceil(memory_gb * 0.9)
+
+    command <<<
+        gatk \
+            --java-options "-XX:GCTimeLimit=50 -XX:GCHeapFreeLimit=10 -Xms4000m -Xmx~{java_heap_size}g" \
+            BaseRecalibratorSpark \
+            -R ~{fasta} \
+            -I ~{bam} \
+            ~{if use_original_quality_scores then "--use-original-qualities" else "" } \
+            -O ~{outfile_name} \
+            -known-sites ~{dbSNP_vcf} \
+            -known-sites ~{sep=" --known-sites " known_indels_sites_VCFs} \
+            --spark-master local[~{ncpu}]
+    >>>
+
+    output {
+        File recalibration_report = "~{outfile_name}"
+    }
+
+    runtime {
+        cpu: ncpu
+        memory: "~{memory_gb} GB"
+        disk: "~{disk_size_gb} GB"
+        container: "quay.io/biocontainers/gatk4:4.4.0.0--py36hdfd78af_0"
+        maxRetries: 1
+    }
+}
+
+task apply_bqsr {
```

Comment:
Just an idea: does it make sense to combine this and the prior task into one? Is there any context where you would run `base_recalibrator` but **not** `apply_bqsr`?

Reply:
It probably would be fine and likely the use case ~100% of the time. However, if someone is used to GATK, I think it may be confusing to only have one step where they expect two. Also, I suspect if the resource requirements were optimized, you'd get different node sizes, so you could probably save costs.